students/markusiwerbo/final_project_code/final_project_code.ino

```diff
@@ -0,0 +1,102 @@
+#include <LiquidCrystal.h>
+
+char dist[4];   //Character array for serial input 
+int nbytes = 0;   
+
+const int numReadings = 3;  // Last three reading are averaged to account for bad measurements 
+float readings[numReadings]; // Array to store readings
+int index = 0;  // Current index in the array
+float total = 0; // Variable to store the sum of readings
+
+//---------------Calibrate the distance sensor----------------//
+const float calibration = 1.05;   // Readings are 5% low 
+//------------------------------------------------------------//
+
+const int rs = 12, en = 11, d4 = 5, d5 = 4, d6 = 3, d7 = 2;
+LiquidCrystal lcd(rs, en, d4, d5, d6, d7);
+
+
+//--------LED Preamble-------//
+int ledpin = 13;
+float slow = 20.0;
+float medium = 15.0;
+float fast = 10.0;
+int ledState = LOW;  // ledState used to set the LED
+unsigned long previousMillis = 0;  // will store last time LED was updated
+
+// -----Blink intervals--------//
+const long interval_s = 600;
+const long interval_m = 1000;
+const long interval_f = 1000;
+
+
+
+void setup() {
+  
+  lcd.begin(16, 1); //set up the LCD's number of columns and rows
+  lcd.print("ArduinoConnected");
+  delay(3000);
+  Serial.begin(9600);
+  for (int i = 0; i < numReadings; i++) {
+    readings[i] = 0;
+    pinMode(ledpin, OUTPUT);
+  }
```

Comment:
This loop seems odd. Should probably set the default `0` value when you instantiate the `readings` array in the preamble. No need to repeatedly set the pinMode for `ledpin`.